app/v2/controllers/requestParsers/validators/TriggerBSASValidator.scala

```diff
@@ -38,28 +36,26 @@ class TriggerBSASValidator @Inject()(val currentDateProvider: CurrentDateProvide
 
   private def bodyFormatValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
     List(
-      JsonFormatValidation.validate[TriggerBsasRequestBody](data.body.json, RuleIncorrectOrEmptyBodyError)
+      JsonFormatValidation.validate[TriggerBsasRequestBody](data.body.json)
     )
   }
 
   private def dateFieldValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
-    val startDate: JsLookupResult = data.body.json \ "accountingPeriod" \ "startDate"
-    val endDate: JsLookupResult = data.body.json \ "accountingPeriod" \ "endDate"
     List(
-      JsonValidation.validate(startDate)(DateValidation.validate(StartDateFormatError)),
-      JsonValidation.validate(endDate)(DateValidation.validate(EndDateFormatError))
+      DateValidation.validateOpt(StartDateFormatError)((data.body.json \ "accountingPeriod" \ "startDate").asOpt[String]),
+      DateValidation.validateOpt(EndDateFormatError)((data.body.json \ "accountingPeriod" \ "endDate").asOpt[String])
     )
   }
 
   private def selfEmploymentIdValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
     List(
-      JsonValidation.validate(data.body.json \ "selfEmploymentId")(SelfEmploymentIdValidation.validate)
+      SelfEmploymentIdValidation.validateOption(data.body.json.\("selfEmploymentId").asOpt[String])
     )
   }
 
   private def typeOfBusinessValidator: TriggerBsasRawData => List[List[MtdError]] = { data =>
     List(
-      JsonValidation.validate(data.body.json \ "typeOfBusiness")(TypeOfBusinessValidation.validate)
+      TypeOfBusinessValidation.validateOption(data.body.json.\("typeOfBusiness").asOpt[String]),
```

Comment:
I really don't like how the validation is done in here... normally, we'd have a `parameterFormatValidation`, `parameterRuleValidation`, `incorrectOrEmptyBodyValidation`, `bodyFormatValidation` and `bodyRuleValidation` (or something like that). Having all of these as separate validations (`dateFieldValidator`, `selfEmploymentIdValidator` etc) is kinda gross, it's making it much harder to tell what's going on in here. I'll go through and refactor some other time, just making a note that this is naaaasty.

Reply:
Also, please replace `.\(...)`

Reply:
Okay a refactor would be beneficial. Also`.\` does the same as ` \ ` but will change if that's an issue